From 11205437be9a199591e0dd72dab5054d31369898 Mon Sep 17 00:00:00 2001
From: Kai-Heng Feng <kai.heng.feng@canonical.com>
Date: Tue, 4 Aug 2020 00:20:02 +0800
Subject: [PATCH 1/2] alsa-mixer: Recognize USB audio jack mixer

UAC v2 and v3 support insertion control (jack detection), and the
created jack mixers have "- Input" suffix and "- Output" suffix for
input jack and output jack, respectively.

Add these jacks so we don't always need to rely on UCM or PulseAudio
profile-set.
---
 src/modules/alsa/mixer/paths/analog-input-linein.conf      | 3 +++
 src/modules/alsa/mixer/paths/analog-input-mic.conf         | 3 +++
 src/modules/alsa/mixer/paths/analog-output-headphones.conf | 3 +++
 src/modules/alsa/mixer/paths/analog-output-speaker.conf    | 3 +++
 4 files changed, 12 insertions(+)

diff --git a/src/modules/alsa/mixer/paths/analog-input-linein.conf b/src/modules/alsa/mixer/paths/analog-input-linein.conf
index 8163ffb2c..cf2079074 100644
--- a/src/modules/alsa/mixer/paths/analog-input-linein.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-linein.conf
@@ -28,6 +28,9 @@ required-any = any
 state.plugged = unknown
 state.unplugged = unknown
 
+[Jack Line - Input]
+required-any = any
+
 [Element Capture]
 switch = mute
 volume = merge
diff --git a/src/modules/alsa/mixer/paths/analog-input-mic.conf b/src/modules/alsa/mixer/paths/analog-input-mic.conf
index 123439b6c..53c03c820 100644
--- a/src/modules/alsa/mixer/paths/analog-input-mic.conf
+++ b/src/modules/alsa/mixer/paths/analog-input-mic.conf
@@ -29,6 +29,9 @@ required-any = any
 state.plugged = unknown
 state.unplugged = unknown
 
+[Jack Mic - Input]
+required-any = any
+
 [Element Capture]
 switch = mute
 volume = merge
diff --git a/src/modules/alsa/mixer/paths/analog-output-headphones.conf b/src/modules/alsa/mixer/paths/analog-output-headphones.conf
index d2147c50f..d4ad7777d 100644
--- a/src/modules/alsa/mixer/paths/analog-output-headphones.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-headphones.conf
@@ -52,6 +52,9 @@ state.unplugged = unknown
 [Jack Headphone Mic]
 required-any = any
 
+[Jack Headphone - Output]
+required-any = any
+
 [Element Hardware Master]
 switch = mute
 volume = merge
diff --git a/src/modules/alsa/mixer/paths/analog-output-speaker.conf b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
index 6f9968e1f..27a3983d5 100644
--- a/src/modules/alsa/mixer/paths/analog-output-speaker.conf
+++ b/src/modules/alsa/mixer/paths/analog-output-speaker.conf
@@ -69,6 +69,9 @@ required-any = any
 state.plugged = unknown
 state.unplugged = unknown
 
+[Jack Speaker - Output]
+required-any = any
+
 [Element Hardware Master]
 switch = mute
 volume = merge
-- 
2.17.1

